test/arkane/encorr/isodesmicTest.py

```diff
@@ -154,55 +153,69 @@ def test_initializing_constraint_map(self):
         }
 
         no_rings = SpeciesConstraints(self.caffeine, [self.butane, self.benzene], conserve_ring_size=False)
-        assert set(no_rings.constraint_map.keys()) == {"H", "C", "O", "N", "C=O", "C-N", "C-H", "C=C", "C=N", "C-C"}
-
-        atoms_only = SpeciesConstraints(self.caffeine, [self.butane], conserve_ring_size=False, conserve_bonds=False)
-        assert set(atoms_only.constraint_map.keys()) == {"H", "C", "O", "N"}
+        caffeine_features = no_rings._get_all_constraints(self.caffeine)
+        caffeine_constraint_list = [feat.__repr__() for feat in caffeine_features]
+        assert set(caffeine_constraint_list) == {"C=O", "C-N", "C-H", "C=C", "C=N", "C-C"}
 
     def test_enumerating_constraints(self):
         """
         Test that a SpeciesConstraints object can properly enumerate the constraints of a given ErrorCancelingSpecies
         """
         spcs_consts = SpeciesConstraints(self.benzene, [])
-        assert set(spcs_consts.constraint_map.keys()) == {"C", "H", "C=C", "C-C", "C-H", "6_ring"}
-
-        # Now that we have confirmed that the correct keys are present, overwrite the constraint map to set the order
-        spcs_consts.constraint_map = {
-            "H": 0,
-            "C": 1,
-            "C=C": 2,
-            "C-C": 3,
-            "C-H": 4,
-            "6_ring": 5,
-        }
+        benzene_features = spcs_consts._get_all_constraints(self.benzene)
+        benzene_constraint_list = [feat.__repr__() for feat in benzene_features]
+        assert set(benzene_constraint_list) == {"C=C", "C-C", "C-H", "6_ring"}
+
+        target_constraints, _ = spcs_consts._enumerate_constraints([benzene_features])
+        benzene_constraints = target_constraints
 
         assert np.array_equal(
-            spcs_consts._enumerate_constraints(self.propene),
-            np.array([6, 3, 1, 1, 6, 0]),
+            benzene_constraints,
+            np.array([1, 3, 6, 3]),
         )
+
+        spcs_consts.all_reference_species = [self.propene]
```

Comment:
Why this is needed?

Reply:
This is to ensure that `reference_constraints` in the code below will only return one thing and so make the check more explicit
```
        spcs_consts.all_reference_species = [self.propene]
        propene_features = spcs_consts._get_all_constraints(self.propene)
        _, reference_constraints = spcs_consts._enumerate_constraints([benzene_features, propene_features])
```